test/unit/public-key-store.js

```diff
@@ -0,0 +1,609 @@
+
+                publicKeyStore.obtainToken(personaClient, testOpts, (err, publicKey) => {
+                    personaClient.tokenCache.get.calledOnce.should.equal(true);
+
+                    personaClient.tokenCache.get.calledWith(testObtainTokenCacheKey).should.equal(true);
+
+                    personaClient.debug.calledTwice.should.equal(true);
+
+                    personaClient.debug.getCall(0).args[0].should.equal('Found cached token for key ' + testObtainTokenCacheKey + ': ' + testPublicKeyString);
+
+                    personaClient.debug.getCall(1).args[0].should.equal('New expires_in is -10');
+                    
+                    personaClient._removeTokenFromCache.calledOnce.should.equal(true);
+
+                    personaClient._removeTokenFromCache.calledWith(testOpts.id, testOpts.secret).should.equal(true);
+
+                    generateTokenStub.called.should.equal(false);
+
+                    err.should.equal(error);
+                    done(); 
+                });
+            });
+        });
+
+        describe('not found in cache', function() {
+            it('empty response from cache', function(done) {
+                var generateTokenStub = sinon.stub();
+                var publicKeyStore = proxyquire('../../client/lib/public-key-store', {
+                    './generate-token': generateTokenStub, 
+                });
+               
+                testPublicKeyObject.expires_at = testExpiredTimestamp;
+        
+                var testPublicKeyString = JSON.stringify(testPublicKeyObject);
+
+                personaClient._formatCacheKey.returns(testObtainTokenCacheKey);
+
+                personaClient.tokenCache.get.callsFake((cacheKey, callback) => {
+                    callback(null, null);
+                });
+
+                generateTokenStub.callsFake((client, cacheKey, xRequestId, id, secret, callback) => {
+                    callback(null, {
+                        access_token: testPublicKeyObject.access_token,
+                        expires_at: testUnexpiredTimestamp,
+                    });
+                });
+
+                publicKeyStore.obtainToken(personaClient, testOpts, (err, publicKey) => {
+                    personaClient.tokenCache.get.calledOnce.should.equal(true);
+
+                    personaClient.tokenCache.get.calledWith(testObtainTokenCacheKey).should.equal(true);
+
+                    generateTokenStub.calledOnce.should.equal(true);
+
+                    generateTokenStub.calledWith(personaClient, testObtainTokenCacheKey, testXRequestId, testOpts.id, testOpts.secret).should.equal(true);
+
+                    publicKey.access_token.should.equal(testPublicKeyObject.access_token);
+                    publicKey.expires_at.should.equal(testUnexpiredTimestamp);
+                    done();
+                });
+            });
+            
+            // https://github.com/talis/platform/issues/1729
+            it('cache error', function(done) {
```

Comment:
This is the test for the functional change in this PR. If the redis cache has an error, fallback to getting the token from Persona.